Document ES6 Modules import

brettz9 opened the pull request:
Note that I didn't try to address a preexisting issue that while `sinonTest` is the global exported in the browser, the sample code uses `sinonTestFactory`. I'm thinking you may want the Node example to use the global to avoid confusion or to explain this somehow (I presume this is easier than renaming the global!).

Also, I think Sinon itself should indicate how to include the script tag or do the import/require, but such a section is completely missing at https://sinonjs.org/#get-started .

fatso83 commented on README.md:
```diff
@@ -57,9 +57,25 @@ via [npm (node package manager)](https://github.com/npm/npm)
-or just add it as a `<script src="dist/sinon-test.js"></script>`
-tag to the html where you write your tests. A pre-built browser
-version can be found in the NPM package under `dist/sinon-test.js`.
+or just add `dist/sinon-test.js` (or for ES6 Modules,
```
These two lines can be removed. They only confuse.

fatso83 commented on README.md:
```diff
@@ -57,9 +57,25 @@ via [npm (node package manager)](https://github.com/npm/npm)
+
+## Browser usage
+
+In place of the `require` statements indicated below, in the
```
I'd place this below the other section (and reference _above_). Something like

# Usage 

## Node and CommonJS build systems
..
## Direct browser usage
..

brettz9 replied:
Sounds good, though Usage would be at heading level 2 to follow the current pattern...

brettz9 commented:
Right, that's what I was referring to--for users of the browser global looking at the sample Node code and adapting it for the browser, there's a little more reconciling to do...

As far as naming, I like `test` better--I think it is enough and conveys a greater lightness to the library...

fatso83 commented on README.md:
````diff
@@ -57,9 +57,25 @@ via [npm (node package manager)](https://github.com/npm/npm)
+only need to add an import statement:
+
+```js
+import sinonTestFactory from './sinon-test/node_modules/dist/sinon-test-es.js'
````
Two things:
1. Error in your import
2. I'd expand the example to include the script tag

Fix:
```html
<script type="module">
import sinon from './node_modules/sinon/pkg/sinon-esm.js';
import sinonTest from './node_modules/sinon-test/dist/sinon-test-es.js';
const test = sinonTest(sinon);

it('should work', test( function() { 
    pass();
}));
</script>
```

fatso83 replied:
P.S. Maybe you could update the examples below in the same manner:
- sinonTestFactory => sinonTest
- sinonTest => test

Then at least the code would align with the usage when coding directly in the browser.

brettz9 replied:
Sounds good--do you want me to also change the existing `var`s in the examples to `let` and `const`?

fatso83 replied:
No, I only think they make sense in an ES2015 setting, as we are otherwise targetting ES5.1.

brettz9 commented:
I've replaced the old commit, taking into account your points.

Note that `const` previously existed in the API section which I didn't change (and, as per your own example that I incorporated, I am using `const` with the direct browser use example (which I assume should be ok since it works in modern browsers)). Let me know if you want either of these changed.

mroderick commented:
Looks good to me 👍
